src/clients/kinit/kinit.c

```diff
@@ -738,63 +697,61 @@ k5_kinit(opts, k5)
         krb5_get_init_creds_opt_set_canonicalize(options, 1);
     if (opts->anonymous)
         krb5_get_init_creds_opt_set_anonymous(options, 1);
-    if (opts->addresses)
-    {
-        krb5_address **addresses = NULL;
-        code = krb5_os_localaddr(k5->ctx, &addresses);
-        if (code != 0) {
-            com_err(progname, code, _("getting local addresses"));
+    if (opts->addresses) {
+        ret = krb5_os_localaddr(k5->ctx, &addresses);
+        if (ret) {
+            com_err(progname, ret, _("getting local addresses"));
             goto cleanup;
         }
         krb5_get_init_creds_opt_set_address_list(options, addresses);
     }
     if (opts->no_addresses)
         krb5_get_init_creds_opt_set_address_list(options, NULL);
-    if (opts->armor_ccache)
-        krb5_get_init_creds_opt_set_fast_ccache_name(k5->ctx, options, opts->armor_ccache);
+    if (opts->armor_ccache != NULL) {
+        krb5_get_init_creds_opt_set_fast_ccache_name(k5->ctx, options,
+                                                     opts->armor_ccache);
+    }
     if (opts->request_pac)
         krb5_get_init_creds_opt_set_pac_request(k5->ctx, options, TRUE);
     if (opts->not_request_pac)
         krb5_get_init_creds_opt_set_pac_request(k5->ctx, options, FALSE);
 
 
-    if ((opts->action == INIT_KT) && opts->keytab_name)
-    {
+    if (opts->action == INIT_KT && opts->keytab_name != NULL) {
 #ifndef _WIN32
         if (strncmp(opts->keytab_name, "KDB:", 4) == 0) {
-            code = kinit_kdb_init(&k5->ctx,
-                                  krb5_princ_realm(k5->ctx, k5->me)->data);
-            if (code != 0) {
-                com_err(progname, code,
+            ret = kinit_kdb_init(&k5->ctx, k5->me->realm.data);
```

Comment:
krb5_princ_realm() is a macro, and I've always found that those macros obfuscate code with no benefit, especially since they take an unused context argument.  We've removed uses of those macros in the past (e.g. caaf72893a5be61822763eb471f4d573992479ed ).